region/lowernorfair/east/Lower Norfair Fireflea Room.json

```diff
@@ -294,6 +294,116 @@
         "Roll to the right wall then the left wall and place a Power Bomb, exit G-Mode, and jump repeatedly to fix the camera."
       ]
     },
+    {
+      "link": [1, 3],
+      "name": "R-Mode Spark Interrupt (Gain Blue Suit) - Crystal Flash",
+      "entranceCondition": {
+        "comeInWithRMode": {}
+      },
+      "requires": [
+        "h_CrystalFlashForReserveEnergy",
+        {"canShineCharge": {"usedTiles": 21, "gentleUpTiles": 3, "gentleDownTiles": 3, "openEnd": 1}},
+        {"autoReserveTrigger": {"maxReserveEnergy": 95}},
+        "canRModeSparkInterrupt"
+      ],
+      "flashSuitChecked": true,
+      "blueSuitChecked": true,
+      "note": [
+        "Crystal Flash, then pick up a shinecharge.",
+        "Drop into the top Fune's fireball path for the interupt."
+      ]
+    },
+    {
+      "link": [1, 3],
+      "name": "R-Mode Spark Interrupt (Gain Blue Suit) - Farm Firefleas",
```

Comment:
I'd like if this were cleaner.
- If Supers or PBs -> farm a fune
- Tricky jump and can jump over bottom fune -> farm a fireflea

Getting past the top rock is a thing, but rolling over it is free.  With trickyJump as knowledge check.
And if you don't have morph, pseudoscrew it, with trickyjump as a knowledge check.